packages/desktop-client/src/hooks/useSplitsExpanded.tsx

```diff
@@ -4,18 +4,76 @@ import React, {
   useEffect,
   useContext,
   useReducer,
+  type Dispatch,
+  type ReactNode,
 } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
 
-const SplitsExpandedContext = createContext(null);
+import {
+  type SplitMode,
+  type SplitState,
+} from 'loot-core/client/state-types/app';
+
+type ToggleSplitAction = {
+  type: 'toggle-split';
+  id: string;
+};
+
+type OpenSplitAction = {
+  type: 'open-split';
+  id: string;
+};
+
+type CloseSplitsAction = {
+  type: 'close-splits';
+  ids: string[];
+};
+
+type SetModeAction = {
+  type: 'set-mode';
+  mode: SplitMode;
+};
+
+type SwitchModeAction = {
+  type: 'switch-mode';
+  id: string;
+};
+
+type FinishSwitchModeAction = {
+  type: 'finish-switch-mode';
+};
+
+type Actions =
+  | ToggleSplitAction
+  | OpenSplitAction
+  | CloseSplitsAction
+  | SetModeAction
+  | SwitchModeAction
+  | FinishSwitchModeAction;
+
+type SplitsStateContext = {
+  state: SplitState;
+  dispatch: Dispatch<Actions>;
+};
+
+const SplitsExpandedContext = createContext<SplitsStateContext>({
+  state: {
+    mode: 'collapse',
+    ids: new Set(),
+    transitionId: null,
+  },
+  dispatch: () => {
+    throw new Error('Unitialised context method called: dispatch');
+  },
+});
 
 export function useSplitsExpanded() {
   const data = useContext(SplitsExpandedContext);
 
   return useMemo(
     () => ({
       ...data,
-      expanded: id =>
+      expanded: (id: string) =>
```

Comment:
I would like to update this to `isExpanded` to make it clear that is returns a boolean
```suggestion
      isExpanded: (id: string) =>
```

Feel free to implement it here and update the references or we could take do it in a separate PR

Reply:
great idea!